rust/query/typeql_insert.rs

```diff
@@ -35,6 +45,43 @@ impl TypeQLInsert {
     }
 }
 
+impl Validatable for TypeQLInsert {
+    fn validate(&self) -> Result<()> {
+        collect_err(
+            &mut [
+                expect_non_empty(&self.variables),
+                expect_insert_in_scope_of_match(&self.match_query, &self.variables),
+            ]
+            .into_iter()
+            .chain(self.match_query.iter().map(Validatable::validate))
+            .chain(self.variables.iter().map(Validatable::validate)),
+        )
+    }
+}
+
+fn expect_insert_in_scope_of_match(
+    match_query: &Option<TypeQLMatch>,
+    variables: &[ThingVariable],
+) -> Result<()> {
+    if let Some(match_query) = match_query {
+        let bounds = match_query.conjunction.names();
```

Comment:
Can we clarify (maybe with the Core team) what is the difference between "bounds" and "scope"?

"scope" is definitely ambiguous, and everywhere we use "bounds" it feels like scope would be appropriate too...